lib/viewcomponents/vertex.jsx

```diff
@@ -41,20 +41,29 @@ var Vertex = React.createClass({
     var vertexWidth = styles.get('width');
     var vertexHeight = styles.get('height');
 
+    var titleWidth = vertexWidth - padding * 2;
+
     var classes = 'vertex shape-' + styles.get('shape');
 
     return (
       <g
-        className={'vertex shape-' + styles.get('shape')} 
+        className={'vertex shape-' + styles.get('shape')}
         transform={'translate(' + position.x + ', ' + position.y + ')'} >
           <path
             d={roundedRectanglePath(0, 0, vertexWidth, vertexHeight, borderRadius)}
             className='vertex-box'
             onMouseDown={mouseDownDrag.bind(this, 'vertex_body', null, null, this._onVertexBodyPseudoDrag)} />
           <path d={roundedRectanglePath(0, 0, vertexWidth, 5, borderRadius, borderRadius, 0, 0)} className="color-bar" fill={styles.get('color')}/>
-          <text ref="titleText" className='label' textAnchor='start' x={padding} y={titlePosition}>
-            {/*model.get('id')*/}
-          </text>
+          {
+            React.createElement('foreignObject', {
+              x: padding,
+              y: titlePosition,
+              width: titleWidth,
+              height: '20',
```

Comment:
Idk. Do you want it to wrap or ellipsis? I'm not sure I know how to make the port labels avoid the vertex label if it is arbitrary height, but that's a minor problem.